src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,595 @@
+		day += 32 - t.Day()
+		month--
+	}
+	if month < 0 {
+		month += 12
+		year--
+	}
+
+	array = append(array, year)
+	array = append(array, month)
+	array = append(array, day)
+	array = append(array, hour)
+	array = append(array, min)
+	array = append(array, sec)
+
+	return
+}
+
+//
+// The following block of code was taken from
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L259
+//Helper function for time difference between two time.Time objects
+func timeDiffHelper(timeList []int) (dif string) {
+	for ind := range timeList {
+		if timeList[ind] > 0 {
+			switch ind {
+			case 0:
+				dif += strconv.Itoa(timeList[ind]) + "Y "
+				break
+			case 1:
+				dif += strconv.Itoa(timeList[ind]) + "M "
+				break
+			case 2:
+				dif += strconv.Itoa(timeList[ind]) + "D "
+				break
+			case 3:
+				dif += strconv.Itoa(timeList[ind]) + "h "
+				break
+			case 4:
+				dif += strconv.Itoa(timeList[ind]) + "m "
+				break
+			case 5:
+				dif += strconv.Itoa(timeList[ind]) + "s "
+				break
+			default:
+				break
+			}
+		}
+	}
+
+	return
+}
+
+//
+// The following block of code was taken from
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L291
+//Function for time difference between two time.Time objects
+func timeDiff(t1 time.Time, t2 time.Time, duration int) string {
+	if t1.After(t2) {
+		timeList := diff(t1, t2)
+		return timeDiffHelper(timeList) + "AGO"
+	}
+
+	timeList := diff(t2, t1)
+	if duration == 1 {
+		return timeDiffHelper(timeList)
+	}
+	return "IN " + timeDiffHelper(timeList)
+
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/alert.go#L426
+//Function for printing annotations in Plain text format
+func tabulate(data []annotationData) {
+
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 8, 8, 0, '\t', 0)
+	defer w.Flush()
+
+	fmt.Fprintf(w, "\n ")
+	for _, each := range configJSON.Columns {
+		fmt.Fprintf(w, "%s\t", each)
+	}
+	fmt.Fprintf(w, "\n")
+
+	for _, each := range data {
+
+		fmt.Fprintf(w, " %d\t%d\t%s\t%s\t%s",
+			each.ID,
+			each.DashboardID,
+			strings.Split(each.Text, "\n<a")[0],
+			each.Tags,
+			timeDiff(time.Now(), time.Unix(0, each.Time*int64(microSec)), 0), // times should be in microseconds of Unix since epoch
+		)
+		if each.TimeEnd == maxtstmp {
+			fmt.Fprintf(w, "\t%s", "Undefined")
+			fmt.Fprintf(w, "\t%s\n", "Undefined")
+		} else {
+			fmt.Fprintf(w, "\t%s", timeDiff(time.Now(), time.Unix(0, each.TimeEnd*int64(microSec)), 0))                                // times should be in microseconds of Unix since epoch
+			fmt.Fprintf(w, "\t%s\n", timeDiff(time.Unix(0, each.Time*int64(microSec)), time.Unix(0, each.TimeEnd*int64(microSec)), 1)) // times should be in microseconds of Unix since epoch
+		}
+	}
+}
+
+//Function running all logics
+func run() {
+
+	var aData []annotationData
+	tagList := parseTags(tags)
+	err := getAnnotations(&aData, tagList)
```

Comment:
I think this should go under default case in switch clause below

Reply:
We would require data for deleting annotations. So I have moved the tabulate() function to the default one. Which basically prints all found annotation in tabular form on stdout.